tests/operations/binary/Add.cpp

```diff
@@ -1,11 +1,43 @@
 #define CATCH_CONFIG_MAIN
 #include "Catch/single_include/catch2/catch.hpp"
 
+#include "sgpl/algorithm/execute_core.hpp"
+#include "sgpl/hardware/Core.hpp"
 #include "sgpl/operations/binary/Add.hpp"
+#include "sgpl/program/Program.hpp"
+#include "sgpl/spec/Spec.hpp"
+
+// define libray and spec
+using library_t = sgpl::OpLibrary<sgpl::Add>;
+using spec_t = sgpl::Spec<library_t>;
+
+// create peripheral
+spec_t::peripheral_t peripheral;
 
 TEST_CASE("Test Add") {
 
-  // TODO flesh out stub test
-  sgpl::Add{};
+  sgpl::Program<spec_t> program{1};
+
+  sgpl::Core<spec_t> core;
+
+  // set up initial state
+
+  // set up values to add in register
+  core.registers[0] = 1;
+  core.registers[1] = 2;
+
+  // set up what registers to operate on
+  program[0].args[0] = 2;
```

Comment:
Could you assign from an emp::array?

Reply:
Fix elsewhere

Reply:
even better, could we make a factory function for the program that has the program written out as a json Rstring (and do this for all of the 1-instruction programs).

Reply:
Still need to assign from emp::array

Reply:
Fix in other files